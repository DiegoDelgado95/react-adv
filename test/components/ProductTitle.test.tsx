import React from 'react'
import renderer from 'react-test-renderer'
import { ProductTitle, ProductCard } from '../../src/components';
import { product1 } from '../data/products';

describe('ProductTittle', () => {
  
    test('debe de mostrar el component con el titulo personalizado', () => { 
        const wrapper = renderer.create(
            <ProductTitle title="Custom Product" className="costume-class" />
        )
        expect( wrapper.toJSON() ).toMatchSnapshot()
     })

     test('debe de mostrar el componente con el nombre del producto', () => {
        const wrapper = renderer.create(
            <ProductCard product={product1}>
                {
                    () => (
                        <ProductTitle />
                    )
                }
            </ProductCard>
        )

        expect( wrapper.toJSON() ).toMatchSnapshot();
      })
});