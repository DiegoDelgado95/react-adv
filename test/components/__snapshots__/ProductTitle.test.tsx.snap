// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductTittle debe de mostrar el component con el titulo personalizado 1`] = `
<span
  className="productDescription costume-class"
>
  Custom Product
</span>
`;

exports[`ProductTittle debe de mostrar el componente con el nombre del producto 1`] = `
<div
  className="productCard undefined"
>
  <span
    className="productDescription undefined"
  >
    Coffe Mug - Card
  </span>
</div>
`;
